pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -461,10 +461,15 @@ func (c *Controller) runDeleteShootFlow(ctx context.Context, o *operation.Operat
 			waitUntilExtensionResourcesDeleted,
 			waitUntilContainerRuntimeResourcesDeleted,
 		)
+		destroyKubeAPIServerSNI = g.Add(flow.Task{
+			Name:         "Destroying kube-apiserver-sni",
+			Fn:           flow.TaskFn(botanist.Shoot.Components.ControlPlane.KubeAPIServerSNI.Destroy).RetryUntilTimeout(defaultInterval, defaultTimeout).DoIf(useSNI),
```

Comment:
Is `DoIf(useSNI)` needed? This call should be executed regardless of the state of the SNI.

Reply:
I just wanted to be sure as I thought that the resources are not created if SNI is disabled. Maybe this was the wrong assumption.

Should I just remove the `DoIf` block?

Reply:
I would say to always remove those resource - why? Because they might have been created when `SNI` was enabled, but during this deletion reconcilation the `SNI` is disabled.

Reply:
Resolved with new commit.